java/src/main/native/src/TableJni.cpp

```diff
@@ -327,9 +550,47 @@ JNIEXPORT void JNICALL Java_ai_rapids_cudf_Table_writeParquet(JNIEnv* env, jclas
     compression_type compression{static_cast<compression_type>(j_compression)};
     statistics_freq stats{static_cast<statistics_freq>(j_stats_freq)};
 
-    cudf::table_view *tview = reinterpret_cast<cudf::table_view *>(j_table);
-    write_parquet_args args(sink, *tview, &metadata, compression, stats);
-    write_parquet(args);
+    write_parquet_chunked_args args(sink, &metadata, compression, stats);
+    std::shared_ptr<detail::parquet::pq_chunked_state> state = write_parquet_chunked_begin(args);
+    cudf::jni::native_parquet_writer_handle * ret =
+        new cudf::jni::native_parquet_writer_handle(state);
+    return reinterpret_cast<jlong>(ret);
+  } CATCH_STD(env, 0)
+}
+
+JNIEXPORT void JNICALL Java_ai_rapids_cudf_Table_writeParquetChunk(JNIEnv* env, jclass,
+    jlong j_state, jlong j_table, jlong mem_size) {
+  JNI_NULL_CHECK(env, j_table, "null table", );
+  JNI_NULL_CHECK(env, j_state, "null state", );
+
+  using namespace cudf::experimental::io;
+  cudf::table_view *tview = reinterpret_cast<cudf::table_view *>(j_table);
+  cudf::jni::native_parquet_writer_handle * state =
+      reinterpret_cast<cudf::jni::native_parquet_writer_handle *>(j_state);
+
+  if (state->sink.get() != nullptr) {
+    long alloc_size = 1024 * 1024 * 10; // 10 MB minimum
+    if (alloc_size < mem_size/2) {
+      alloc_size = mem_size/2;
+    }
+    state->sink->set_alloc_size(alloc_size);
+  }
+  try {
+    write_parquet_chunked(*tview, state->state);
+  } CATCH_STD(env, )
+}
+
+JNIEXPORT void JNICALL Java_ai_rapids_cudf_Table_writeParquetEnd(JNIEnv* env, jclass,
+    jlong j_state) {
+  JNI_NULL_CHECK(env, j_state, "null state", );
+
+  using namespace cudf::experimental::io;
+  cudf::jni::native_parquet_writer_handle * state =
+      reinterpret_cast<cudf::jni::native_parquet_writer_handle *>(j_state);
+  try {
+    write_parquet_chunked_end(state->state);
+    // TODO should we do something to be 100% sure this is deleted???
```

Comment:
Asserting the `use_count` might not be a bad idea if we are positive no other pointers should be sharing the state at this point.

Reply:
I was thinking more along the idea of exception handling and wanting to be 100% sure that we delete even if `write_parquet_chunked_end` throws.